examples/platforms/simulation/virtual_time/platform-sim.c

```diff
@@ -70,16 +71,20 @@ static void handleSignal(int aSignal)
     gTerminate = true;
 }
 
-void otSimSendEvent(const struct Event *aEvent)
+void otSimSendEvent(struct Event *aEvent)
 {
     ssize_t            rval;
     struct sockaddr_in sockaddr;
-
     memset(&sockaddr, 0, sizeof(sockaddr));
     sockaddr.sin_family = AF_INET;
     inet_pton(AF_INET, "127.0.0.1", &sockaddr.sin_addr);
     sockaddr.sin_port = htons(9000 + sPortOffset);
-
+#if OPENTHREAD_SIMULATION_EXT_RF_MODELS
+    // fill in V2 message format default fields (same for all events).
+    aEvent->mEventV2Indicator = OT_SIM_EVENT_V2_FORMAT;
+    aEvent->mNodeId           = gNodeId;
```

Comment:
The node ID was added some time ago for the integration with the Whitefield simulator. The aim/hope was that we can use the same OT simulation binary for potentially multiple simulators; and not all simulators use the same approach of using different UDP port numbers. For example it could use another communication method / pipe that doesn't have port numbers and still we like to use the same event format in this case.  Containing all the relevant information into the event data itself then is very useful.

Reply:
That make sense to me. I was actually trying to see if it's possible to use a faster IPC mechanism than sockets.

If so, maybe we can add the nodeid for all simulators. 
Just prefer to not add excess configurations when they are not fundamentally necessary.

Reply:
One other note on this topic; for debug purposes I will soon  introduce a "message ID" variable to verify per OT node what is the last message from the simulator that it has seen. This is likely not functionally required but should help with the debug and validation process. Can then be removed later on.